source/core/parameters_lagrangian.cc

```diff
@@ -317,6 +318,16 @@ namespace Parameters
           {
             std::runtime_error("Invalid integration method");
           }
+
+        const std::string insertion = prm.get("insertion_method");
+        if (insertion == "uniform")
+          insertion_method = InsertionMethod::uniform;
+        else if (insertion == "non_uniform")
+          insertion_method = InsertionMethod::non_uniform;
+        else
+          {
+            std::runtime_error("Invalid insertion method");
+          }
```

Comment:
This does not belong here. The insertion method should be in the insertion sub section, not the DEM model one.